### PR TITLE
doc: input: add two paragraphs about keymap and hid mapping

### PR DESCRIPTION
**Pull request by fabiobaltieri**
Add a paragraph about the input-keymap binding and one about the HID mapping functions.

### CODE REVIEW

File: doc/services/input/gpio-kbd.rst [hunk 1]
```diff
@@ -199,6 +199,39 @@ The actual key mask can be changed at runtime by enabling
...
+              MATRIX_KEY(2, 1, INPUT_KEY_8)
+              MATRIX_KEY(2, 2, INPUT_KEY_9)
+          >;
+          row-size = <3>;
+          col-size = <3>;
+      };
+  };
+
+Would setup a ``keymap`` device that take the x/y/touch events as an input and
+generate corresponding key events as an output.
```
Comment (kartben) [inline doc/services/input/gpio-kbd.rst:233]: I would move this before - "For example, the following would setup [...]"

File: doc/services/input/index.rst [hunk 2]
```diff
@@ -55,6 +55,14 @@ footprint, or in a complex application with an existing event model, where the
 callback is just a wrapper to pipe back the event in a more complex application
 specific event system.
 
+HID code mapping
+****************
+
+A common use case for input devices is to use them to generate HID reports. For
+this purpose the :c:func:`input_to_hid_code` and
```
Comment (kartben) [inline doc/services/input/index.rst:62]:
```suggestion
this purpose, the :c:func:`input_to_hid_code` and
```

File: doc/services/input/index.rst [hunk 3]
```diff
@@ -55,6 +55,14 @@ footprint, or in a complex application with an existing event model, where the
...
+:c:func:`input_to_hid_modifier` can be used to map input codes to HID codes and
```
Comment (kartben) [inline doc/services/input/index.rst:63]:
```suggestion
:c:func:`input_to_hid_modifier` functions can be used to map input codes to HID codes and
```

**Review by kartben — CHANGES_REQUESTED** [review]

File: doc/services/input/gpio-kbd.rst (hunk 1 above)
Comment (fabiobaltieri) [reply doc/services/input/gpio-kbd.rst]: done

File: doc/services/input/index.rst (hunk 2 above)
Comment (fabiobaltieri) [reply doc/services/input/index.rst]: done

File: doc/services/input/index.rst (hunk 3 above)
Comment (fabiobaltieri) [reply doc/services/input/index.rst]: and done

File: doc/services/input/gpio-kbd.rst [hunk 4]
```diff
@@ -199,6 +199,38 @@ The actual key mask can be changed at runtime by enabling
 :kconfig:option:`CONFIG_INPUT_KBD_ACTUAL_KEY_MASK_DYNAMIC` and the using the
 :c:func:`input_kbd_matrix_actual_key_mask_set` API.
 
+Keymap configuration
+********************
+
+Keyboard matrix devices report a series of x/y/touch events, these can be
+mapped to normal key events using the :dtcompatible:`input-keymap` driver.
+
+For example, the following Would setup a ``keymap`` device that take the
```
Comment (henrikbrixandersen) [inline doc/services/input/gpio-kbd.rst:208]:
```suggestion
For example, the following would setup a ``keymap`` device that take the
```

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Looks good, two minor nits:

File: doc/services/input/gpio-kbd.rst [hunk 5]
```diff
@@ -199,6 +199,38 @@ The actual key mask can be changed at runtime by enabling
 :kconfig:option:`CONFIG_INPUT_KBD_ACTUAL_KEY_MASK_DYNAMIC` and the using the
 :c:func:`input_kbd_matrix_actual_key_mask_set` API.
 
+Keymap configuration
+********************
+
+Keyboard matrix devices report a series of x/y/touch events, these can be
```
[the remark this answers is not in the document]
Comment (fabiobaltieri) [reply doc/services/input/gpio-kbd.rst]: thanks, fixed

File: doc/services/input/gpio-kbd.rst (hunk 4 above)
Comment (fabiobaltieri) [reply doc/services/input/gpio-kbd.rst]: fixed

**Review by henrikbrixandersen — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**